core/sdk-core/src/main/java/software/amazon/awssdk/core/ResponseBytes.java

```diff
@@ -30,13 +28,16 @@
  * {@link ResponseTransformer#toBytes()} or {@link AsyncResponseTransformer#toBytes()} to a streaming output operation.
  */
 @SdkPublicApi
-public final class ResponseBytes<ResponseT> {
+public final class ResponseBytes<ResponseT> extends SdkBytes {
```

Comment:
What's the benefit of making it extend `SdkBytes`? It doesn't seem to be a `is-a` relation between those two and Inheritance might introduce issues, eg: 
- We cannot make `SdkBytes` final and it's a public API
- We cannot make field `responseBytes` private 
- `ResponseBytes` class will have all the static methods in `SdkBytes`, but most of `fromXXX` might be useless since they are returning `SdkBytes`. 

Should we use composition instead of inheritance here?

Reply:
It feels like an is-a relationship to me, but I can understand the concern. 

Composition would be a bigger barrier to the customer using it. We'd probably want to copy+paste the methods and keep them in sync so that it's flattened within ResponseBytes. 

We could do a common base class to remove the copy+paste and static method issues, but it wouldn't fix all of your concerns.

Reply:
+1 to not extending. I'd rather just have this class contain an instance of SdkBytes and delegate to it (hide that from the customer though so they don't have to do responseBytes.sdkBytes().asString()) but instead can go directly to string (responseBytes.asString()). I think it's worth a bit of duplication and if we are delegating anyways it's just duplicating the interfaces really.

Reply:
Added a package-protected base class as a compromise between preventing future issues and hatred of extension. This way, it behaves like composition for all intents and purposes (and can be converted to actually do so in the future, backwards-compatibly), but prevents us from updating it in one place and forgetting to update it in another.